src/ConnectedMode/SlCoreConnectionAdapter.cs

```diff
@@ -91,7 +91,11 @@ public Task<AdapterResponseWithData<List<OrganizationDisplay>>> GetOrganizations
 
             try
             {
-                var credentials = MapCredentials(credentialsModel?.ToICredentials());
+                if (credentialsModel == null)
```

Comment:
In case the credentialsModel is null the ToICredentials will throw a generic null exception. In this case I believe it's better to be proactive and add back the `.?` so the exception we get becomes specific to the lack of credentials.